File: app/src/settings/workshop.ts
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

import type { app_WorkshopSettings } from "@tsukilabs/nil-bindings";

export class WorkshopSettingsImpl implements app_WorkshopSettings {
  public hideUnmet = __CONSTS__.defaultSettings.workshop.hideUnmet;
}

File: crates/nil-payload/src/response/infrastructure/workshop.rs
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

use nil_core::infrastructure::building::workshop::recruit_catalog::WorkshopRecruitCatalog;
use serde::{Deserialize, Serialize};

#[cfg(feature = "axum")]
use nil_payload_macros::IntoJsonResponse;

#[cfg(feature = "typescript")]
use ts_rs::TS;

#[derive(Clone, Debug, Deserialize, Serialize)]
#[cfg_attr(feature = "axum", derive(IntoJsonResponse))]
#[cfg_attr(feature = "typescript", derive(TS))]
#[cfg_attr(feature = "typescript", ts(export))]
pub struct GetWorkshopRecruitCatalogResponse(pub WorkshopRecruitCatalog);
